tests/models/test_dag.py

```diff
@@ -2125,6 +2132,60 @@ def test_dags_needing_dagruns_not_too_early(self):
         session.rollback()
         session.close()
 
+    def test_dags_needing_dagruns_datasets(self, session):
+        dataset = Dataset(uri='hello')
+        dag = DAG(
+            dag_id='my_dag', max_active_runs=1, schedule=[dataset], start_date=pendulum.now().add(days=-2)
+        )
+        EmptyOperator(task_id='dummy', dag=dag)
+        DAG.bulk_write_to_db([dag], session=session)
+        session.commit()
```

Comment:
(Will need re-formating) By using the `dag_maker` fixture it will automatically clean up after the test:

```suggestion
    def test_dags_needing_dagruns_datasets(self, dag_maker, session):
        dataset = Dataset(uri='hello')
        with dag_maker(session=session,dag_id='my_dag', max_active_runs=1, schedule=[dataset], start_date=pendulum.now().add(days=-2)
        ):
            EmptyOperator(task_id='dummy')
```